synapseclient/core/credentials/credential_provider.py

```diff
@@ -62,10 +71,16 @@ class UserArgsSessionTokenCredentialsProvider(SynapseCredentialsProvider):
     """
 
     def _get_auth_info(self, syn, user_login_args):
+        username = None
+        password = None
+        api_key = None
+        auth_token = None
+
         if user_login_args.session_token:
-            return syn.getUserProfile(sessionToken=user_login_args.session_token)['userName'], None,\
-                   syn._getAPIKey(user_login_args.session_token)
-        return None, None, None
+            username = syn.getUserProfile(sessionToken=user_login_args.session_token)['userName']
+            api_key = syn._getAPIKey(user_login_args.session_token)
```

Comment:
i wonder if we should request a Access token instead of the API key. we can do this in a future issue.

Reply:
Yes, I'd like to do this also but it's a bit trickier because it would mean generating a new access token with each rememberMe login, each of which would be saved, and which are limited to 100, so it requires some additional interaction to support if happening automatically on a login. I'm thinking it should probably be a discrete step rather than happening automatically on login.